CustomCraftSML/PublicAPI/CustomCraft.cs

```diff
@@ -60,141 +60,295 @@ public static bool AddEntry<T>(T entry)
-            // Value chosen for what should be the standard inventory size
+            if (craftingTab.FabricatorType == CraftTree.Type.None)
+            {
+                QuickLogger.Error($"Error on crafting tab '{craftingTab.TabID}'. ParentTabPath must identify a fabricator for the custom tab.");
+                return false;
+            }
+            try
+            {
+                HandleCraftingTab(craftingTab);
 
-            CraftDataHandler.SetItemSize(GetTechType(customSize.ItemID), customSize.Width, customSize.Height);
+                return true;
+            }
+            catch (Exception ex)
+            {
+                QuickLogger.Error($"Exception thrown while handling crafting tab '{craftingTab.TabID}'{Environment.NewLine}{ex}");
+                return false;
+            }
         }
 
-        internal static void CustomizeBioFuel(ICustomBioFuel customBioFuel)
+        internal static bool MoveRecipe(IMovedRecipe movedRecipe)
         {
-            Assert.IsTrue(GetTechType(customBioFuel.ItemID) <= TechType.Databox, "This API in intended only for use with standard, non-modded TechTypes.");
+            if (!movedRecipe.IsComplete)
+            {
+                QuickLogger.Warning($"Error on moved recipe for '{movedRecipe.ItemID}'. Moved recipes require both the Old and New paths.");
+                return false;
+            }
+
+            var oldPath = new CraftingPath(movedRecipe.OldPath);
+            var newPath = new CraftingPath(movedRecipe.NewPath);
+
+            string[] oldSteps = (oldPath.Path + CraftingNode.Splitter + movedRecipe.ItemID).Split(CraftingNode.Splitter);
+            string[] newSteps = newPath.Path.Split(CraftingNode.Splitter);
 
-            BioReactorHandler.SetBioReactorCharge(GetTechType(customBioFuel.ItemID), customBioFuel.Energy);
+            CraftTreeHandler.RemoveNode(oldPath.Scheme, oldSteps);
+
+            TechType itemID = GetTechType(movedRecipe.ItemID);
+
+            if (newSteps.Length <= 1)
+                CraftTreeHandler.AddCraftingNode(newPath.Scheme, itemID);
+            else
+                CraftTreeHandler.AddCraftingNode(newPath.Scheme, itemID, newSteps);
+
+            return true;
         }
 
-        internal static void CustomCraftingTab(ICraftingTab craftingTab)
+        internal static bool CustomizeFragments(ICustomFragmentCount fragments)
         {
-            Assert.IsTrue(craftingTab.SpriteItemID <= TechType.Databox, "This API in intended only for use with standard, non-modded TechTypes.");
-            Assert.IsTrue(craftingTab.FabricatorType <= CraftTree.Type.Rocket, "This API in intended only for use with standard, non-modded CraftTree.Types.");
-            Assert.IsTrue(craftingTab.FabricatorType > CraftTree.Type.None, "ParentTabPath must identify a fabricator for the custom tab.");
+            TechType itemID = GetTechType(fragments.ItemID);
+
+            if (itemID == TechType.None)
+                return false;
```

Comment:
Fragments only truly exist for a handful of existing game items.  
Add additional checks here:
- Quick check: `TechType` should not be a modded techtype
- Longer check: Confirm if the provided `TechType` is a fragment itself or is for a blueprint unlocked by fragment scanning.